internal/githubrepository/controller.go

```diff
@@ -88,6 +87,10 @@ func (c *Controller) Run() error {
 
 	log.Printf("Starting controller...")
 
+	if c.cfg.ReconciliationPeriodSeconds > 0 {
```

Comment:
Is there a reason you've allowed a 0 period?

Would it make sense to force the controller to have a resync period - do this check in the command initiation?